qiita_db/util.py

```diff
@@ -926,11 +985,13 @@ def get_processed_params_tables():
     -------
     list of str
     """
-    sql = ("SELECT * FROM information_schema.tables WHERE table_schema = "
-           "'qiita' AND SUBSTR(table_name, 1, 17) = 'processed_params_'")
-
-    conn = SQLConnectionHandler()
-    return sorted([x[2] for x in conn.execute_fetchall(sql)])
+    with TRN:
+        sql = """SELECT table_name FROM information_schema.tables
+                 WHERE table_schema = 'qiita'
+                    AND SUBSTR(table_name, 1, 17) = 'processed_params_'
+                 ORDER BY table_name"""
```

Comment:
I was thinking on edge cases with non-alphanumeric characters ... but
this is referring to table names, so I think it's all good!

On (Jul-03-15|17:23), josenavas wrote:
> > @@ -926,11 +985,13 @@ def get_processed_params_tables():
> >      -------
> >      list of str
> >      """
> > -    sql = ("SELECT * FROM information_schema.tables WHERE table_schema = "
> > -           "'qiita' AND SUBSTR(table_name, 1, 17) = 'processed_params_'")
> > -
> > -    conn = SQLConnectionHandler()
> > -    return sorted([x[2] for x in conn.execute_fetchall(sql)])
> > +    with TRN:
> > +        sql = """SELECT table_name FROM information_schema.tables
> > +                 WHERE table_schema = 'qiita'
> > +                    AND SUBSTR(table_name, 1, 17) = 'processed_params_'
> > +                 ORDER BY table_name"""
> 
> I think so, why shouldn't they?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1324/files#r33883417